### ISSUE TITLE
New skulpt.org homepage

### ISSUE THREAD

**Issue by eah13** [skulpt/skulpt#454]
Anyone interested in helping revamp the skulpt.org homepage?  @bnmnetp called for a PRP and I'm happy to take this on and would love help and suggestions.

**Broad brush thoughts:**

* use github pages (i.e. a new repo) instead of including within the skulpt source. I've [created a custom domain](https://help.github.com/articles/setting-up-a-custom-domain-with-github-pages/) for a github pages site for my [personal site](http://elliotthauser.com) so happy to help with this.  It's pretty painless.  Note that I do believe we should create an organization page with a custom domain instead of a project page within a gh-pages branch in this repo.  That will be simpler for development and accepting pull requests.
* Responsive framework.  [Bootstrap](http://getbootstrap.com/) and [Zurb Foundation](http://foundation.zurb.com/) seem to be the two standards.  They make it easy to make mobile-friendly sites with lots of nice built-in functionality.  I'd lean towards Foundation since I have more experience with it unless someone else loves bootstrap and wants to volunteer to do lots of work. They're similar enough that experience transfers very easily.
* Jekyll. This seems to be the lingua franca of static sites.  Certainly light templating will be important.  There's [Pelican](http://blog.getpelican.com/) in Python but honestly Jekyll has been pretty good to me even though I don't know Ruby so I'd strongly prefer Jekyll. Liquid is a decent enough templating language to do most of what we'll need of it.

**Content.**  I think at first the content can stay the same while we make the large changes above.  But once it's transferred, we'll be able to have some organization and light nav, so themed pages will make sense.  Seems like the main division might be amongst types of audience:

* For everyone: What skulpt is, and a hands-on place to play with it.  See also Try Skulpt
* For learners: How to use skulpt to learn.  See also the Gallery
* For teachers: How to use skulpt to teach.  See also the Gallery
* For site owners: How to integrate skulpt into your site. See also usage docs
* For OSS developers: How to get involved in core Skulpt development.  See also dev docs.

**New Sections:**

* Try Skulpt.  I love how the current page has an interactive example front-and-center.  It certainly is what drew me in the first time I landed on the page.  We should definitely keep that focus. I'd suggest this as the home page.  Idea from @LeszekSwirski: examples that call out awesome or unique features of Skulpt like suspensions, turtle, and matplotlib.
* Project Gallery: CodeSkulptor, Runestone, Trinket, GeometryZen, and the other great projects that use Skulpt.  Perhaps indicating whether they're for students, teachers, or both.  Helps showcase the maturity of the project and that it's used in production currently.  The current site has some links which are great but I'm thinking a top-level page of its own. PRs could add new projects here.  Perhaps a randomized order on pageload so every project gets a fair share of the spotlight :)
* Usage Docs: How to add the latest version of Skulpt to your site and hook it up to a text area & run button.  This should include links to preferred CDN(s).  Maybe eventually a cloneable example jekyll site. This section may be the place to highlight performance characteristics, suspensions, and other things that matter in production.
* Dev Docs: Our current and future materials on how to contribute and how Skulpt works.

Finally, here are some nice-to-haves that likely require further discussion:

* Mission statement?  Skulpt is a cross-browser Python interpreter for education.  Some statement like that will help people grok what's going on and how the project differs from pypy.js (a fast, complete Javascript JIT compiler for distributed system), Brython (Aiming to replace Javascript as the Web's scripting language), or Rapydscript (allowing a sugary Pythonic way to write Javascript, like Coffeescript).  Python, Client Side is nice and succinct but can be confusing in light of these other projects.
* Logo? This is mostly fluff, but it does seem like a great logo or mascot can help a project mature.  At the very least it would let me buy a t-shirt from Cafe Press :)

What did I forget?  

Let's keep this issue in a brainstorming mode ("yes, and" rather than "no") for now.  Once we've gotten the basics hammered out and everyone's had a chance to say they're interested I can create a new repo (skulpt.github.io) and add contributors and we'll dig into details from there.  Open to other paths forward, of course, but this is what made sense to my brain.

**Comment by LeszekSwirski** [comment]
I'm definitely all for keeping that interactive example right there on the landing page, it's definitely a "wow" moment that you can use it immediately, especially on mobile.

**Comment by LeszekSwirski** [comment]
I'm biased, but I would add suspensions to the live demo, e.g. tracing execution, allowing pausing, or having a long-running process which doesn't block the browser thread.

**Comment by eah13** [comment]
@LeszekSwirski I like that. I'll add something about examples that demonstrate key features

**Comment by eah13** [comment]
Here's a rough hack to show how existing content would fit into foundation.

Code:
https://github.com/eah13/skulpt-redo/

Live site:
http://eah13.github.io/skulpt-redo/

Make your browser window skinny to see the responsive juju in action.

If that looks good, I'll need @bnmnetp to create a repo titled `skulpt.github.io` (I can't create repos) and give me commit access.  I can then hack in the open there.

**Comment by bnmnetp** [comment]
OK, Its there and you have write permissions.

1.  I think the home page can be shorter -- less scrolling with links to pages for some of the major sections.

2.  I would like to see documentation play a more prominent role on the skulpt page.  I think some combination of HACKING and CONTRIBUTING and gists would go a ways.   If we can get a framework for documenting things in place then we can probably do much better at filling in details.

**Comment by eah13** [comment]
Lovely!

The new in-progress site is up:
http://skulpt.github.io/

Pull requests welcome:
https://github.com/skulpt/skulpt.github.io/

@bnmnetp agreed that documention should be a first-class, top-level citizen in the new site.  Will try to make some to-do issues and give a nav framework that reflects this.

Closing this issue- anyone who's interested please join me in the new repo.

**Comment by LeszekSwirski** [comment]
I'm a bit uncomfortable that this website got pushed through with so little discussion, and that the issue got closed so quickly. I think the discussion on the `.gitattributes` file, #451, lasted longer and got more input from more people than the discussion for this new website. Plus, the new website definitely has sections that I'm not comfortable with (see https://github.com/skulpt/skulpt.github.io/issues/4).

**Comment by bnmnetp** [comment]
@LeszekSwirski, 

Good points, but check out http://skulpt.github.io/skulpt

@eah13 moved the site back onto the gh-pages branch of the main skulpt repo, this is a better way to do it than in a separate repository.  I do not believe that you will see trinket.io on the front page.  There is a page to highlight projects as you suggest.

Also, this is a work in progress and does not replace skupt.org until we create a CNAME file and I give scott the go ahead to change the DNS record to point to the new site.  I think we have much more to talk about in terms of getting some really compelling examples on the front page and getting our documentation in order before we call this done.

Brad

**Comment by eah13** [comment]
Agreed on all counts. Happy to have lots of discussion and input here.

The draft site is here: http://skulpt.github.io/skulpt 
Code here: https://github.com/skulpt/skulpt/tree/gh-pages

As of now it's essentially just a wireframe as you can see from the [placeholder images on the gallery page](http://skulpt.github.io/skulpt/gallery.html).  Yesterday I cut up index.html and put the content where it seemed to make sense.  I haven't added any content myself, really, besides stubbing out some project gallery projects to make sure they work.  I envision the entire list of projects that use skulpt making it into that gallery.  The Featured Project at the top, by the way, is chosen randomly at build time so every merge to gh-pages will get a new random featured project onto the site as github runs Jekyll on their end.  Someone with more JS skills than I could make it be chosen randomly at page load, which would be even better.

RE: Trinket on skulpt.org: The trinket currently on skulpt.org was [added by @bnmnetp](https://github.com/skulpt/skulpt/blame/master/doc/index.html#L165) about a year ago.  We've appreciated the publicity and have gotten a lot of awareness and users from it.  10-15% of site visitors interact with or use it in some way, even though it's way at the bottom, so I think @LeszekSwirski's idea of putting it somewhere like the Using Skulpt section makes most sense. The actual index page should be all about Try Skulpt, and very focused.

**Comment by LeszekSwirski** [comment]
My sincerest apologies for the misplaced blame regarding the Trinket section, I have no excuse other than a late evening and tired eyes. I must have thought the old page was the new one, or something of the sort.

**Comment by rixner** [comment]
I think it's reasonable to show people that they can use Trinket to get code onto their site.  However, I will say that I find it a bit odd that the first thing you see on the page entitled "Using Skulpt" is something telling me to use Trinket.  Then, after that, it shows you how to actually use Skulpt itself.  I feel like the order should be flipped, with it showing you how to use Skulpt first.  Then saying something about how Trinket can help if you just want to embed editable code in your website (or whatever you guys would like to say about Trinket).

**Comment by eah13** [comment]
@rixner agreed on all counts. I only cut and pasted the content into what seemed like the appropriate place, with no thought to order. I'll start cleaning up the pages this week. I'll do pull requests for each of these major sets of changes so we can do comments and review there.

While we're on the topic, I think the Using Skulpt page would probably start with buttons that would zoom visitors to the section they were interested in. One of those sections could be trinket, the other may be using Skulpt from a CDN, downloading skulpt-dist, and maybe using bower. Comments welcome- that's all the thinking I've done about it yet. I'll try to do roughly an hour a day of work on the site. 

If anyone wants to help out let me know and we can coordinate (and I'll stop just pushing to gh-pages like I've been doing in the setup phase). The gh-pages branch has a readme with setup and basic hacking instructions.

**Comment by eah13** [comment]
@LeszekSwirski no worries! I figured there might be some crossed wires. It's partially my fault since the new site development so far has been kind of tough for others to follow. Please keep the constructive feedback coming.

**Comment by rixner** [comment]
@eah13, that seems reasonable.  I agree that there needs to be some sort of table of contents, as the page is likely to have a lot of content on how to use Skulpt.

**Comment by eah13** [comment]
PR #467 has some more work on using.html.  For now I've also got a live version of that PR up on my fork at http://elliotthauser.com/skulpt/using.html for convenience of review.

Hope to see you on #467 if you've got any comments.
